apollo-federation/src/sources/connect/json_selection/apply_to.rs

```diff
@@ -85,15 +99,15 @@ impl Hash for ApplyToError {
 }
 
 impl ApplyToError {
-    fn new(message: &str, path: &[JSON]) -> Self {
+    pub(crate) fn new(message: &str, path: Vec<JSON>) -> Self {
```

Comment:
```suggestion
    pub(crate) fn new(message: String, path: Vec<JSON>) -> Self {
```

Since we need the owned value anyway, we should take it as input. That way, if the caller _has_ the ownership, we're not double-cloning.

Reply:
https://github.com/apollographql/router/pull/5762/commits/d73bb4d15aa2ce22bd251a4b3746f22262b27571